### PR TITLE
Improve comments in migrations/example.exyaml

### PR DESCRIPTION
**Pull request by hmaarrfk**
Checklist
* [ ] Used a [fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

File: recipe/migrations/example.exyaml [hunk 1]
```diff
@@ -1,9 +1,16 @@
...
+  # Only use the migration_number if the bot messes up,
+  # putting this in will cause a complete rerun of the migration
+  migration_number:
+    1
+
+# The name of the feedstock you wish to migrate
 python:
-  - 3.6
-  - 3.8a1
+  - 3.6    # old version
```
Comment (CJ-Wright) [inline recipe/migrations/example.exyaml]: This is not necessarily true, in this case we're building 3.6 and 3.8a1. The value here is whatever you want the resulting config to be.

File: recipe/migrations/example.exyaml [hunk 2]
```diff
@@ -1,9 +1,16 @@
...
+# The timestamp of when the migration was made
+# Can be obtained by copying the output of
+# python -c "import time; print(f'{time.time():.0f}')"
+migrator_ts: 0
+__migrator:
+  kind:
     version
-  migration_number:  # Only use this if the bot messes up, putting this in will cause a complete rerun of the migration
-    1 
+  # Only use the migration_number if the bot messes up,
```
Comment (CJ-Wright) [inline recipe/migrations/example.exyaml]:
```suggestion
  # Only change the migration_number if the bot messes up,
```

**Comment by CJ-Wright** [comment]
you might add an xref to https://github.com/conda-forge/conda-forge-enhancement-proposals/blob/master/cfep-09.md

File: recipe/migrations/example.exyaml [hunk 3]
```diff
@@ -1,9 +1,18 @@
...
+  # Only change the migration_number if the bot messes up,
+  # putting this in will cause a complete rerun of the migration
+  migration_number:
+    1
+
+# The name of the feedstock you wish to migrate
 python:
-  - 3.6
-  - 3.8a1
+  - 3.6    # list of new versions
```
Comment (CJ-Wright) [inline recipe/migrations/example.exyaml]:
```suggestion
  - 3.6    # list of versions to build against
```

**Comment by hmaarrfk** [comment]
Where does the new pinning get set?

**Comment by CJ-Wright** [comment]
What do you mean exactly?
Conda smithy has the ability to merge all the pinning migration yaml files and the global config before it creates the variants so that the effective config is the composite.

For the most part the values in the pinning yaml override the values in the global config where they are in conflict. In the example we update the python versions we build against to 3.6 and 3.8a1 since we specified the python key and gave those two versions. For a simple version bump migration one would provide a yaml with the new version, like done in #296

**Comment by hmaarrfk** [comment]
Right, it is just unclear to me what a list of versions means for a pinning. But I kinda get it

**Comment by CJ-Wright** [comment]
A list of versions mean that we build against all the listed versions as a build matrix for instance [python](https://github.com/conda-forge/conda-forge-pinning-feedstock/blob/master/recipe/conda_build_config.yaml#L545)

**Comment by CJ-Wright** [comment]
I updated the example to use boost rather than python, since single version migrations are much more common than build matrix migrations. The full spec for running complex migrations is in the CFEP9 proposal itself.

File: recipe/migrations/example.exyaml [hunk 4]
```diff
@@ -1,9 +1,17 @@
...
+__migrator:
+  kind:
     version
-  migration_number:  # Only use this if the bot messes up, putting this in will cause a complete rerun of the migration
-    1 
-python:
-  - 3.6
-  - 3.8a1
+  # Only change the migration_number if the bot messes up,
+  # putting this in will cause a complete rerun of the migration
```
Comment (CJ-Wright) [inline recipe/migrations/example.exyaml]:
```suggestion
  # changing this will cause a complete rerun of the migration
```

**Comment by CJ-Wright** [comment]
Thank you very much!

**Comment by hmaarrfk** [comment]
No problem. Thanks for your help!

**Pull request merged**